File: source/includes/fts-tutorial/partial-match/autocomplete-query.go
package main

import (
	"context"
	"fmt"
	"time"

	"go.mongodb.org/mongo-driver/bson"
	"go.mongodb.org/mongo-driver/mongo"
	"go.mongodb.org/mongo-driver/mongo/options"
)

// define structure of movies collection
type MovieCollection struct {
	title string `bson:"Title,omitempty"`
	plot  string `bson:"Plot,omitempty"`
}

func main() {
	var err error
	// connect to the Atlas cluster
	ctx := context.Background()
	client, err := mongo.Connect(ctx, options.Client().ApplyURI("<connection-string>"))
	if err != nil {
		panic(err)
	}
	defer client.Disconnect(ctx)
	// set namespace
	collection := client.Database("sample_mflix").Collection("movies")
	// define pipeline
	searchStage := bson.D{{"$search", bson.M{
		"autocomplete": bson.M{
			"path": "plot", "query": "new purchase", "tokenOrder": "any", "fuzzy": bson.M{
				"maxEdits": 2, "prefixLength": 1, "maxExpansions": 256},
		},
		"highlight": bson.D{
			{"path", "plot"},
		},
	}}}
	limitStage := bson.D{{"$limit", 5}}
	projectStage := bson.D{{"$project", bson.D{{"title", 1}, {"plot", 1}, {"_id", 0}, {"highlights", bson.D{{"$meta", "searchHighlights"}}}}}}
	// specify the amount of time the operation can run on the server
	opts := options.Aggregate().SetMaxTime(5 * time.Second)
	// run pipeline
	cursor, err := collection.Aggregate(ctx, mongo.Pipeline{searchStage, limitStage, projectStage}, opts)
	if err != nil {
		panic(err)
	}
	// print results
	var results []bson.D
	if err = cursor.All(context.TODO(), &results); err != nil {
		panic(err)
	}
	for _, result := range results {
		fmt.Println(result)
	}
}

File: content/search/source/includes/partial-match-tutorial/code-snippets/rust/autocomplete-query.rs
use mongodb::bson::{doc, DateTime, from_document};
use mongodb::{Client, Collection};
use serde::{Deserialize, Serialize};
use futures::TryStreamExt;

#[derive(Debug, Deserialize, Serialize)]
struct Movie {
    title: Option<String>,
    plot: Option<String>,
    genres: Option<Vec<String>>,
    released: Option<DateTime>,
}

#[tokio::main]
async fn main() -> mongodb::error::Result<()> {
    let uri = "<connection-string>";
    let client = Client::with_uri_str(uri).await?;
    
let agg = vec![
        doc! {
            "$search": {
                "index": "partial-match-tutorial",
                "autocomplete": {
                    "path": "plot",
                    "query": "new purchase",
                    "tokenOrder": "any",
                    "fuzzy": {
                        "maxEdits": 2,
                        "prefixLength": 1,
                        "maxExpansions": 256
                    }
                },
                "highlight": {
                    "path": "plot"
                }
            }
        },
        doc! {
            "$limit": 5
        },
        doc! {
            "$project": {
                "_id": 0,
                "title": 1,
                "plot": 1,
                "highlights": {
                    "$meta": "searchHighlights"
                }
            }
        }
    ];

    let coll: Collection<Movie> = client
        .database("sample_mflix")
        .collection("movies");
    while let Some(doc) = result.try_next().await? {
        let movie: Movie = from_document(doc)?;
        println!("  Title: {}", movie.title.unwrap_or_else(|| "N/A".to_string()));
        println!("  Plot: {}", movie.plot.unwrap_or_else(|| "N/A".to_string()));
        println!();
    }

    Ok(())
}